230 us additional views navigation

witnick opened the pull request:
I abstracted away the schema selector in the top bar and the one in the setting into a single component( see SchemaSelector.tsx) since they have the same behaviour.

I also changed the layout of the accordion that display the views at Eddie's request. It is no longer showing the views based on the table but instead it is based on the schema. 

Small demo:
![additionalview](https://github.com/WillTrem/UInnovate/assets/10947924/140f6c11-2b3c-4b04-bec6-87f9daee33d5)

I created an AdditionalViewNavBar component. When a table has additional views available it will show with the different options, otherwise it is hidden. 
For the time being it only shows the available display type and the links don't lead to a view. Future stories will be the implementation of the views

![image](https://github.com/WillTrem/UInnovate/assets/10947924/e6e364c6-fd85-497e-bb54-e0d0134791eb)

AfandiM reviewed (dismissed):
It looks good to me. Given the size of the PR, I would recommend an additional review from someone else, but 👍

lyubageorgieva commented on UInnovateApp/src/pages/Settings.tsx:
```diff
@@ -51,9 +53,9 @@ export function Settings() {
-	const handleSchemaChange = (event: SelectChangeEvent) => {
-		setSelectedSchema(event.target.value);
-	};
+	// const handleSchemaChange = (event: SelectChangeEvent) => {
```
Do you need all of this commented out?

lyubageorgieva commented on UInnovateApp/src/pages/Settings.tsx:
```diff
@@ -77,7 +79,7 @@ export function Settings() {
 						<Box display="flex" gap={"2rem"} alignItems={"center"}>
 							<h1 className='title'>Settings</h1>
-							<FormControl fullWidth disabled={schemas.length === 0}>
+							{/* <FormControl fullWidth disabled={schemas.length === 0}>
```
This also?

lyubageorgieva commented on UInnovateApp/src/virtualmodel/AdditionalViewsDataAccessor.ts:
```diff
@@ -72,7 +82,7 @@ export const insertNewView = async (p_schemaName: string, p_tableName: string, p
 		}
-	
+
 		// alert("View saved successfully.");
```
The comment here also?

lyubageorgieva commented on UInnovateApp/src/components/AdditionalViewNavBar.tsx:
```diff
@@ -0,0 +1,95 @@
+        const signal = ctrl.signal;
+        // get data from db
+        getViewsBySchema(setViewList, selectedSchema, signal);
+        // getCustomViews(setCustomViews, signal)
```
Do we need this?

witnick replied:
yes that is actually the requirement for #230.

lyubageorgieva reviewed (changes requested):
Is it possible to fix the buttons, they are all lower caps while if you look at our other tabs it's all caped, and also with their colours in the modal. They will be a general refactoring later on but if it's possible to fix the letter case at least and the colours from now it would help for later on 
<img width="1252" alt="Screenshot 2024-02-24 at 9 49 53 AM" src="https://github.com/WillTrem/UInnovate/assets/74876532/9cbb6fc9-004e-4dd2-bfc0-2130d09053d2">

lyubageorgieva reviewed (changes requested):
Here I see View three times, and I find it not easy on the eyes.

<img width="224" alt="Screenshot 2024-02-24 at 9 55 56 AM" src="https://github.com/WillTrem/UInnovate/assets/74876532/01eb132d-687b-4564-8ea1-a7e254ea08a5">


Maybe instead, just add ADD VIEW without the title and the "no view"
<img width="780" alt="Screenshot 2024-02-24 at 9 59 18 AM" src="https://github.com/WillTrem/UInnovate/assets/74876532/ae895335-b6ff-4e60-b03f-d6b3814a9894">

lyubageorgieva reviewed (changes requested):
Here also the button "delete"
<img width="1276" alt="Screenshot 2024-02-24 at 10 01 14 AM" src="https://github.com/WillTrem/UInnovate/assets/74876532/413611f5-bb8b-4725-86e9-73de07e2c697">

witnick commented:
I changed the style of the buttons and removed the View title

![additionalview_icons](https://github.com/WillTrem/UInnovate/assets/10947924/8b819286-40c9-4a94-9c53-f16101c55830)
